### DISCUSSION TITLE
Consider using CIS hardened images

### DISCUSSION THREAD

**Discussion by jemrobinson** [alan-turing-institute/data-safe-haven#1246] [Ideas]
We should check whether the [CIS hardened VM images](https://www.cisecurity.org/blog/cis-hardened-images-now-in-microsoft-azure-marketplace/) work with our deployments and if they do, consider the costs/benefits.

**Comment by martintoreilly** [comment]
## CIS hardened images for Safe Haven VMs available on Azure
Source: https://www.cisecurity.org/cis-hardened-image-list/

- [CIS Ubuntu Linux 18.04 LTS Benchmark](https://azuremarketplace.microsoft.com/en-us/marketplace/apps/center-for-internet-security-inc.cis-ubuntu-linux-1804-l1?ref=website)
- [NGINX](https://azuremarketplace.microsoft.com/en-us/marketplace/apps/center-for-internet-security-inc.cis-nginx-centos-7-v1-1-0-l1?ref=website)
- [PostgreSQL](https://azuremarketplace.microsoft.com/en-us/marketplace/apps/center-for-internet-security-inc.cis-postgresql-11-centos-linux-7-level-1?ref=website)
- [Windows 2019 Server](https://azuremarketplace.microsoft.com/en-us/marketplace/apps/center-for-internet-security-inc.cis-windows-server-2019-v1-0-0-l1?ref=website)

## CIS Benchmarks
Source: https://www.cisecurity.org/cis-benchmarks/

These are hardening recommendations, not pre-built images.
- [Apache HTTP Server](https://www.cisecurity.org/benchmark/apache_http_server)
- [Apache Tomcat](https://www.cisecurity.org/benchmark/apache_tomcat)
- [Docker](https://www.cisecurity.org/benchmark/docker)
- [Google Chrome](https://www.cisecurity.org/benchmark/google_chrome)
- [MIT Kerberos](https://www.cisecurity.org/benchmark/mit_kerberos)
- [Microsoft Azure](https://www.cisecurity.org/benchmark/azure)
- [Microsoft IIS](https://www.cisecurity.org/benchmark/microsoft_iis)
- [Microsoft SQL Server](https://www.cisecurity.org/benchmark/microsoft_sql_server)
- [Microsoft Windows Server](https://www.cisecurity.org/benchmark/microsoft_windows_server)
- [MongoDB](https://www.cisecurity.org/benchmark/mongodb)
- [Mozilla Firefox](https://www.cisecurity.org/benchmark/mozilla_firefox)
- [NGINX](https://www.cisecurity.org/benchmark/nginx)
- [Oracle MySQL](https://www.cisecurity.org/benchmark/oracle_mysql)
- [Palo Alto Networks](https://www.cisecurity.org/benchmark/palo_alto_networks)
- [PostgreSQL](https://www.cisecurity.org/benchmark/postgresql)
- [Ubuntu Linux](https://www.cisecurity.org/benchmark/ubuntu_linux)

**Comment by martintoreilly** [comment]
## Bitnami images
Not CIS certified but certified by Bitnami as up-to-date and secure.

- [Gitlab CE](https://bitnami.com/stack/gitlab)
- [NGINX](https://bitnami.com/stack/nginx)
- [Guacamole](https://bitnami.com/stack/guacamole)
- [PostreSQL](https://bitnami.com/stack/postgresql/cloud)
- [MySQL](https://bitnami.com/stack/mysql/cloud)
- [MongoDB](https://bitnami.com/stack/mongodb/cloud)

**Comment by martintoreilly** [comment]
## DoD STIGs
US Department of Defence Security Technical Implementation Guides: https://www.stigviewer.com/stigs

These are hardening recommendations, not pre-built images, and include generic STIGs for e.g. Network Security and Remote Access in addition to STIGs for specific operating systems and software.

**Comment by JimMadge** [comment]
I've had a look at their Ubuntu 18.04 benchmark. Do we know what changes they have implemented in their image? I understand they might not want to protect their work, but it's harder to have confidence when we can't see what they have done.

Some of the recommendations, like having certain 'system' directories on separate partitions, would be hard to replicate ourselves as the default Ubuntu images comes with a standard one-partition layout.
